test/srt/cpu/test_shared_expert.py

```diff
@@ -0,0 +1,223 @@
+import itertools
+import math
+import unittest
+
+import torch
+import torch.nn as nn
+
+# TODO: use interface in cpu.py
+from sgl_kernel.common_ops import convert_weight_packed
+from sgl_kernel.common_ops import shared_expert_cpu as shared_expert
+from utils import (
+    BLOCK_K,
+    BLOCK_N,
+    SiluAndMul,
+    factor_for_scale,
+    fp8_max,
+    fp8_min,
+    per_token_quant_int8,
+    precision,
+    scaled_weight,
+    torch_naive_moe,
+    torch_w8a8_per_column_moe,
+)
+
+from sglang.test.test_utils import CustomTestCase
+
+
+class TestSharedExpert(CustomTestCase):
+    M = [2, 121]
+    N = [32, 32 * 4]
+    K = [32, 32 * 2]
+    routed_scaling_factor = [16]
+
+    M_fp8 = [2, 12]
+    N_fp8 = [128, 256]
+    K_fp8 = [256, 1024]
```

Comment:
I guess here you use two sets of M, N and K because of the BLOCK_SIZE constraint in fp8 block gemm. We can just use one set of M, N, K here. Just make sure that we cover both M <=4 and M >4, as we have different implementations in the underlying kernels.

Reply:
UT has been updated in https://github.com/sgl-project/sglang/pull/6339/commits/16f52038a459810bd7b7db8279209734ad60d5e2 to use one set of N and K. For M, we test M=2 and M=12.